apps/backend/src/reconciliation/reconciliation.service.ts

```diff
@@ -0,0 +1,84 @@
+import { POSDepositEntity } from './entities/pos-deposit.entity';
+import { TransactionEntity } from './entities/transaction.entity';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CashDepositEntity } from './entities/cash-deposits';
+
+@Injectable()
+export class ReconciliationService {
+  constructor(
+    @InjectRepository(TransactionEntity)
+    private salesRepo: Repository<TransactionEntity>,
+    @InjectRepository(POSDepositEntity)
+    private TDI34: Repository<POSDepositEntity>,
+    @InjectRepository(CashDepositEntity)
+    private TDI17: Repository<CashDepositEntity>
+  ) {}
+
+  //TODO ADD DTO AND RETURN TYPE
+  async addSales(body: any[]) {
+    return body.forEach(async (itm: any) => {
+      await this.salesRepo.save(this.salesRepo.create(itm));
+    });
+  }
+
+  //TODO ADD DTO
+  async addTDI34(body: any) {
+    body.details.forEach(async (itm: any) => {
+      return await this.TDI34.save(
+        this.TDI34.create({
+          // location: location ? location : undefined,
+          merchant_terminal: itm.merchant_terminal,
+          terminal_no: itm.terminal_no,
+          // payment_method: payment_method ? payment_method : undefined,
+          card_vendor: itm.card_vendor,
+          card_id: itm.card_id,
+          transaction_date: itm.transaction_date,
+          transaction_time: itm.transaction_time,
+          settlement_date: itm.settlement_date,
+          transaction_cd: itm.transaction_cd,
+          transaction_amt: itm.transaction_amt
+        })
+      );
+    });
+  }
+
+  //TODO ADD DTO AND RETURN TYPE
+  async addTDI17(body: any) {
+    body.details.forEach(async (itm: any) => {
+      return await this.TDI17.save(
+        this.TDI17.create({
```

Comment:
create inside save?

Reply:
This is a shortened version of:

```
const entity = await this.entity.create({entityData})
if (entity){
  await this.entity.save(entity)
}
```

Reply:
The repository pattern in typeorm requires a create and then save. I am not sure why, but I suppose if you wanted to create an entity and maybe perform some other logic on it, or pass it to another function before saving?